woocommerce/google-listings-and-ads
# Fix grouping of shipping rate by price and currency

### Changes proposed in this Pull Request:

Closes #1189.

Prior to this PR, in `getCountriesPriceArray` function, we were grouping shipping rates by price only, and for each of the grouping, the currency was set based on the currency of the first shipping rate. This causes issue #1189. 

In this PR, we fix the issue by grouping the shipping rates by price and currency.

Also, there were two copies of the `getCountriesPriceArray` function. I have refactored them and put the function into utils folder. 

Update: As per https://github.com/woocommerce/google-listings-and-ads/pull/1190#discussion_r783332727 below, I have also renamed the function `getCountriesPriceArray` to `groupShippingRatesByPriceCurrency`.

### Screenshots:

In Setup MC:

![image](https://user-images.githubusercontent.com/417342/149170263-d82e92b1-2f6d-40b4-8dff-dcb7ad08cbe1.png)

In Edit Free Listings:

![image](https://user-images.githubusercontent.com/417342/149176586-c96e7661-bf6f-46e3-b86e-c6ee09e39317.png)

### Detailed test instructions:

1. Go to Setup MC, and proceed to setup a few shipping rates.
2. Go to WooCommerce store settings and change the store currency.
3. Go back to Setup MC and proceed to setup a few more shipping rates. The new shipping rates should display the new store currency.
4. Reload the page. The shipping rates should be displayed with their associated currencies.
5. Complete the Setup MC flow.
6. Go to Dashboard tab and click on the edit button for Free Listings.
7. Go to Step 2. The shipping rates should be displayed with their associated currencies.


### Changelog entry

> Fix - Group shipping rate by price and currency, and display the right currency in shipping rate input.
